packages/roosterjs-editor-plugins/lib/plugins/Paste/Paste.ts

```diff
@@ -74,23 +62,17 @@ export default class Paste implements EditorPlugin {
         if (event.eventType == PluginEventType.BeforePaste) {
             const { htmlAttributes, fragment, sanitizingOption, clipboardData } = event;
             const trustedHTMLHandler = this.editor.getTrustedHTMLHandler();
-            let wacListElements: Node[];
+            let wacListElements: Node[] = [];
 
-            if (isWordDocument(htmlAttributes)) {
+            if (isWordDesktopDocument(htmlAttributes)) {
```

Comment:
A suggestion.

Since all these checks are exclusive to each other, how about we have one unified checker function to return a KnownSourceType, then do switch ... case to handle it? What do you think?

Reply:
Sounds good let me refactor that